src/renderer.ts

```diff
@@ -26,13 +27,11 @@ export function factory (environment: string): ErrorRequestHandler {
   return (err: Boom<ErrorData>, _req: Request, res: Response, _next: NextFunction) => {
     const { message, output: { statusCode: status }, data } = err
 
-    const code = data && data.code
-      ? data.code
-      : slug(err.output.payload.error, { replacement: '_', lower: true })
+    const code = data && data.code ? data.code : slug(err.output.payload.error, { replacement: '_', lower: true })
 
-    const output = shouldDisplayErrorStack(environment) && data && data.stack
-      ? { status, error: { code, message, stack: data.stack }, additionalData: err.data }
-      : { status, error: { code, message }, additionalData: err.data }
+    const output = shouldDisplayErrorStack(environment) && data
+      ? { status, error: { code, message, stack: data.stack, data: data.additionalProperties } }
+      : { status, error: { code, message, data: data!.additionalProperties } }
```

Comment:
Actually, I believe we should have a ternary here to check if data really exists, because it can actually be `null`

So maybe
```ts
    const additionalData = data && data.additionalProperties ? data.additionalProperties : undefined
    const output = shouldDisplayErrorStack(environment) && data
      ? { status, error: { code, message, stack: data.stack, data: additionalData } }
      : { status, error: { code, message, data: additionalData } }
```

Or we could check if data exists in a separate `if` statement, like this:

```ts
const output = { status, error: { code, message } }

if (shouldDisplayErrorStack(environment) && data && data.stack) output.error.stack = data.stack

if (data && data.additionalProperties) output.error.data = data.additionalProperties
```

I think the second one is cleaner